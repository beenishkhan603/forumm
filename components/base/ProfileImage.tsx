import { useState } from 'react'
import User from '@public/images/User.svg'
import Image from 'next/image'

const ProfileImage = ({
  imageUrl,
  className = '',
  onClick,
  size = 'md',
  activityStatus,
  tooltip,
}: {
  imageUrl?: string | null | undefined
  size?: 'xs' | 'sm' | 'md' | 'lg'
  className?: string
  onClick?: (e: any) => void
  activityStatus?: boolean | undefined | null
  tooltip?: string[]
}) => {
  const [profileImageError, setProfileImageError] = useState(false)
  const [showTooltip, setShowTooltip] = useState(false)

  const sizes = {
    xs: 'w-6 h-6',
    sm: 'w-8 h-8',
    md: 'w-10 h-10',
    lg: 'w-16 h-16',
  }

  const handleMouseEnter = () => setShowTooltip(true)
  const handleMouseLeave = () => setShowTooltip(false)

  return (
    <div
      className={`relative flex-shrink-0 ${sizes[size]} ${className} ${
        onClick ? 'cursor-pointer' : ''
      }`}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      {profileImageError || !imageUrl ? (
        <User
          className={`${sizes[size]} rounded-full  bg-white`}
          onClick={onClick}
        />
      ) : (
        <Image
          src={imageUrl}
          width={32}
          height={32}
          alt="Profile Image"
          className={`${sizes[size]} bg-contain bg-center rounded-full object-cover `}
          onError={() => setProfileImageError(true)}
          onClick={onClick}
        />
      )}
      {/* {activityStatus != null && ( */}
      {/*   <div */}
      {/*     className={`w-2.5 h-2.5 rounded-full -mt-2 -mr-0.5 right-0 absolute ${ */}
      {/*       activityStatus */}
      {/*         ? 'bg-green-600 opacity-90' */}
      {/*         : 'bg-gray-500 opacity-70' */}
      {/*     }`} */}
      {/*   ></div> */}
      {/* )} */}
      {tooltip && showTooltip && (
        <div
          style={{ width: '150px' }}
          className="absolute pt-4 text-center bg-gray-700 text-white rounded-md shadow-lg -top-14 left-1/2 transform -translate-x-1/2 -translate-y-1/2 tooltip-arrow"
        >
          <p className="text-md mb-1 font-bold ml-4 mr-4">{tooltip[0]}</p>
          {tooltip.slice(1).map((row) => (
            <p key={row} className="text-xs mb-1 ml-4 mr-4">
              {row}
            </p>
          ))}
          <p></p>
        </div>
      )}
    </div>
  )
}

export default ProfileImage
