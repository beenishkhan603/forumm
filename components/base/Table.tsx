import { useTheme } from '@libs/useTheme'
import { v4 } from 'uuid'

const Table = ({
  tableHeading,
  rows,
}: {
  tableHeading: (JSX.Element | string | number)[]
  rows?: (JSX.Element | string | number)[][]
}) => {
  const { theme } = useTheme()
  const random_id = v4()
  return (
    <table className="w-full table-auto text-left rounded-lg border border-midnight-gray">
      <thead className="">
        <tr
          style={{ backgroundColor: theme.foregroundColour }}
          className="border-b-midnight-gray border-b"
          key={`table-head-${random_id}-container`}
        >
          {tableHeading.map((t) => (
            <th
              className="p-3"
              key={`table-head-${random_id}-${t.toString().length > 0 ? t.toString() : v4()}`}
            >
              {t}
            </th>
          ))}
        </tr>
      </thead>
      <tbody>
        {rows?.map((row, index) => {
          return (
            <tr
              key={`table-row-${random_id}-${index}`}
              className={`py-2 border-b-midnight-gray border-b`}
            >
              {row.map((t, index) => (
                <td className="p-3" key={`table-data-${random_id}-${index}`}>
                  {index === 3 && t === 'Break' ? (
                    <span className="rounded border border-orange-200 text-orange-200 py-0.5 px-1 text-sm">
                      Break
                    </span>
                  ) : (
                    t
                  )}
                </td>
              ))}
            </tr>
          )
        })}
      </tbody>
    </table>
  )
}

export default Table
