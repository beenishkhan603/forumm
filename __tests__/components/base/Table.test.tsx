import ProfileImage from '@components/base/ProfileImage'
import Table from '@components/base/Table'
import { render } from '@testing-library/react'

it('renders table unchanged', () => {
  const { container } = render(
    <Table
      tableHeading={['Name', 'Email', '']}
      rows={[
        ['John', 'john.doe@gmail.com', <div key={1}>Actions</div>],
        ['Pete', 'pete.doe@gmail.com', <div key={2}>Actions</div>],
      ]}
    />
  )
  expect(container).toMatchSnapshot()
})
