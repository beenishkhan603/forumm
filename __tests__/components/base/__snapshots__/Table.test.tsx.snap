// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders table unchanged 1`] = `
<div>
  <table
    class="w-full table-auto text-left rounded-lg border border-midnight-gray"
  >
    <thead
      class=""
    >
      <tr
        class="border-b-midnight-gray border-b"
        style="background-color: rgb(251, 252, 254);"
      >
        <th
          class="p-3"
        >
          Name
        </th>
        <th
          class="p-3"
        >
          Email
        </th>
        <th
          class="p-3"
        />
      </tr>
    </thead>
    <tbody>
      <tr
        class="py-2 border-b-midnight-gray border-b"
      >
        <td
          class="p-3"
        >
          John
        </td>
        <td
          class="p-3"
        >
          john.doe@gmail.com
        </td>
        <td
          class="p-3"
        >
          <div>
            Actions
          </div>
        </td>
      </tr>
      <tr
        class="py-2 border-b-midnight-gray border-b"
      >
        <td
          class="p-3"
        >
          Pete
        </td>
        <td
          class="p-3"
        >
          pete.doe@gmail.com
        </td>
        <td
          class="p-3"
        >
          <div>
            Actions
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
